src/libraries/System.IO.Compression/src/System/IO/Compression/DeflateZLib/DeflateStream.cs

```diff
@@ -793,8 +798,9 @@ internal ValueTask WriteAsyncMemory(ReadOnlyMemory<byte> buffer, CancellationTok
             EnsureNoActiveAsyncOperation();
             EnsureNotDisposed();
 
-            return cancellationToken.IsCancellationRequested ?
-                ValueTask.FromCanceled(cancellationToken) :
+            return
+                cancellationToken.IsCancellationRequested ? ValueTask.FromCanceled(cancellationToken) :
+                buffer.IsEmpty ? default :
```

Comment:
This is a bit hard to read outside of an IDE. Consider adding parentheses around the second inline if else statement.

Reply:
I read this as:
```C#
if (cancellationToken.IsCancellationRequested) return ValueTask.FromCanceled(cancellationToken);
if (buffer.IsEmpty) return default;
return Core(buffer, cancellationToken);
```
Where are you recommending parens go?